runelite-client/src/main/java/net/runelite/client/plugins/slayer/Task.java

```diff
@@ -117,55 +234,188 @@
-	TUROTH("Turoth", ItemID.TUROTH, NpcID.TUROTH, NpcID.TUROTH_427, NpcID.TUROTH_428, NpcID.TUROTH_429, NpcID.TUROTH_430, NpcID.TUROTH_431, NpcID.TUROTH_432),
-	TZHAAR("Tzhaar", ItemID.ENSOULED_TZHAAR_HEAD, NpcID.TZHAARHUR, NpcID.TZHAARHUR_2162, NpcID.TZHAARHUR_2163, NpcID.TZHAARHUR_2164, NpcID.TZHAARHUR_2165, NpcID.TZHAARHUR_2166, NpcID.TZHAARHUR_7682, NpcID.TZHAARHUR_7683, NpcID.TZHAARHUR_7684, NpcID.TZHAARHUR_7685, NpcID.TZHAARHUR_7686, NpcID.TZHAARHUR_7687, NpcID.TZHAARKET, NpcID.TZHAARKET_2174, NpcID.TZHAARKET_2175, NpcID.TZHAARKET_2176, NpcID.TZHAARKET_2177, NpcID.TZHAARKET_2178, NpcID.TZHAARKET_2179, NpcID.TZHAARKET_2186, NpcID.TZHAARKET_2187, NpcID.TZHAARKET_7679, NpcID.TZHAARMEJ, NpcID.TZHAARMEJ_2155, NpcID.TZHAARMEJ_2156, NpcID.TZHAARMEJ_2157, NpcID.TZHAARMEJ_2158, NpcID.TZHAARMEJ_2159, NpcID.TZHAARMEJ_2160, NpcID.TZHAARMEJ_7652, NpcID.TZHAARXIL, NpcID.TZHAARXIL_2168, NpcID.TZHAARXIL_2169, NpcID.TZHAARXIL_2170, NpcID.TZHAARXIL_2171, NpcID.TZHAARXIL_2172),
-	VAMPIRES("Vampires", ItemID.STAKE, NpcID.VAMPYRE_JUVENILE, NpcID.VAMPYRE_JUVENILE_3693, NpcID.VAMPYRE_JUVENILE_3696, NpcID.VAMPYRE_JUVENILE_3697, NpcID.VAMPYRE_JUVENILE_4436, NpcID.VAMPYRE_JUVENILE_4437, NpcID.VAMPYRE_JUVENILE_4438, NpcID.VAMPYRE_JUVENILE_4439, NpcID.VAMPYRE_JUVENILE_8326, NpcID.VAMPYRE_JUVENILE_8327, NpcID.VAMPYRE_JUVINATE, NpcID.VAMPYRE_JUVINATE_3691, NpcID.VAMPYRE_JUVINATE_3694, NpcID.VAMPYRE_JUVINATE_3695, NpcID.VAMPYRE_JUVINATE_3698, NpcID.VAMPYRE_JUVINATE_3699, NpcID.VAMPYRE_JUVINATE_3700, NpcID.VAMPYRE_JUVINATE_4427, NpcID.VAMPYRE_JUVINATE_4428, NpcID.VAMPYRE_JUVINATE_4429, NpcID.VAMPYRE_JUVINATE_4430, NpcID.VAMPYRE_JUVINATE_4432, NpcID.VAMPYRE_JUVINATE_4433, NpcID.VAMPYRE_JUVINATE_4434, NpcID.VAMPYRE_JUVINATE_4442, NpcID.VAMPYRE_JUVINATE_4481, NpcID.VAMPYRE_JUVINATE_4443, NpcID.VAMPYRE_JUVINATE_4482, NpcID.VAMPYRE_JUVINATE_4486, NpcID.VAMPYRE_JUVINATE_4487, NpcID.VAMPYRE_JUVINATE_5634, NpcID.VAMPYRE_JUVINATE_5635, NpcID.VAMPYRE_JUVINATE_5636, NpcID.VAMPYRE_JUVINATE_5637, NpcID.VAMPYRE_JUVINATE_5638, NpcID.VAMPYRE_JUVINATE_5639, NpcID.VAMPYRIC_HOUND, NpcID.FERAL_VAMPYRE, NpcID.FERAL_VAMPYRE_3234, NpcID.FERAL_VAMPYRE_3237, NpcID.FERAL_VAMPYRE_3239, NpcID.FERAL_VAMPYRE_3707, NpcID.FERAL_VAMPYRE_3708, NpcID.FERAL_VAMPYRE_4431, NpcID.FERAL_VAMPYRE_5640, NpcID.FERAL_VAMPYRE_5641, NpcID.FERAL_VAMPYRE_5642, NpcID.FERAL_VAMPYRE_6529, NpcID.FLYING_FEMALE_VAMPIRE, NpcID.FLYING_FEMALE_VAMPIRE_3765, NpcID.FLYING_FEMALE_VAMPIRE_3766, NpcID.FLYING_FEMALE_VAMPIRE_3767, NpcID.FORMER_VAMPYRE, NpcID.FORMER_VAMPYRE_3702, NpcID.FORMER_VAMPYRE_3703, NpcID.FORMER_VAMPYRE_3704, NpcID.FORMER_VAMPYRE_3705, NpcID.FORMER_VAMPYRE_3706),
+	TROLLS("Trolls", ItemID.TROLL_GUARD, NpcID.TROLL, NpcID.TROLL_2833, NpcID.TROLL_8470, NpcID.TROLL_8471,
```

Comment:
This should include the various "greater" trolls (such as Stick and Rock), as well as ice trolls and river trolls.